include/motor-control/core/linear_motion_system.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <concepts>
+
+namespace lms {
+
+template <typename MC>
+concept MotorMechanicalConfig = requires(MC mc) {
+    {mc.mm_per_rev};
+};
+
+struct BeltConfig {
```

Comment:
If we change this struct so that it does not have a user-defined constructor, then it will be a candidate for [aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization) including designated-initialization, which means you could initialize it like
```c++
auto config = BeltConfig{.belt_pitch = 2, .pulley_tooth_count = 4};
```
It's also possible that it would be a [LiteralType](https://en.cppreference.com/w/cpp/named_req/LiteralType) which would allow instances of it to be `constexpr`:
```c++
constexpr auto config = BeltConfig{.belt_pitch=2, .pulley_tooth_count=2};
```
The only thing that might prevent this is the way `mm_per_rev` works, and changing that to a constexpr function as either
- a constexpr member function: `constexpr float mm_per_rev() const { return belt_pitch * pully_count; }`
- a constexpr static member function: `static constexpr float BeltConfig::mm_per_rev(const BeltConfig& bc) const { return bc.belt_pitch * bc.pulley_tooth_count; }`
- a constexpr free function: `constexpr float mm_per_rev(const BeltConfig& bc) { return bc.belt_pitch * bc.pulley_tooth_count; }` (with one version for each of the belt/leadscrew configs)

I think it would be really nice, if ultimately not really difference making, for these to be constexpr-safe.

Reply:
Totally agree with you!